test/libs/oboeAsserter.js

```diff
@@ -16,398 +8,405 @@ if(isNode) {
       }
 
-      return this;
-    };
+      return this
+    }
 
     /** sinon stub is only really used to record arguments given.
      *  However, we want to preserve the arguments given at the time of calling, because they might subsequently
      *  be changed inside the parser so everything gets cloned before going to the stub
      */
-    function argumentClone(delegateCallback) {
-      return function(){
-
-        function clone(original){
-          // Note: window.eval being used here instead of JSON.parse because
-          // eval can handle 'undefined' in the string but JSON.parse cannot.
-          // This isn't wholy ideal since this means we're relying on JSON.
-          // stringify to create invalid JSON. But at least there are no
-          // security concerns with this being a test.
-          return globalContext.eval( '(' + JSON.stringify( original ) + ')' );
+    function argumentClone (delegateCallback) {
+      return function () {
+        function clone (obj) {
+          var copy
+
+          // Handle the 3 simple types, and null or undefined
+          if (obj == null || typeof obj !== 'object') return obj
+
+          if (obj instanceof Error) {
+            return obj
+          }
+
+          // Handle Date
+          if (obj instanceof Date) {
+            copy = new Date()
+            copy.setTime(obj.getTime())
+            return copy
+          }
+
+          // Handle Array
+          if (obj instanceof Array) {
+            copy = []
+            for (var i = 0, len = obj.length; i < len; i++) {
+              copy[i] = clone(obj[i])
+            }
+            return copy
+          }
+
+          // Handle Object
+          if (obj instanceof Object) {
+            copy = {}
+            for (var attr in obj) {
+              if (obj.hasOwnProperty(attr)) copy[attr] = clone(obj[attr])
+            }
+            return copy
+          }
+
+          throw new Error("Unable to copy obj! Its type isn't supported.")
         }
-        function toArray(args) {
-          return Array.prototype.slice.call(args);
+
+        function toArray (args) {
+          return Array.prototype.slice.call(args)
         }
 
-        var cloneArguments = toArray(arguments).map(clone);
+        var cloneArguments = toArray(arguments).map(clone)
 
-        return delegateCallback.apply( this, cloneArguments );
-      };
+        return delegateCallback.apply(this, cloneArguments)
+      }
     }
   }
-  return new OboeAsserter();
+  return new OboeAsserter()
 }
 
-
 var wasPassedAnErrorObject = {
-  testAgainst: function failIfNotPassedAnError(callback, oboeInstance) {
-
-    if( !callback.args[0][0] instanceof Error ) {
-      throw new Error("Callback should have been given an error but was given" + callback.constructor.name);
+  testAgainst: function failIfNotPassedAnError (callback, oboeInstance) {
+    if (!(callback.args[0][0].thrown instanceof Error)) {
```

Comment:
`!callback.args[0][0] instanceof Error` always returns false because `!callback.args[0][0]` resolves to a Boolean. Correcting this issue highlighted the issue with the `clone`.